doc/content/getting-started/api/_index.md

```diff
@@ -0,0 +1,179 @@
+
+## HTTP Queries
+
+Additional fields may be specified in HTTP requests by appending them as query string parameters. For example, to request the `name`, `description`, and `locations` of devices in an `EndDeviceRegistry.Get` request, add these fields to the `field_mask` field. To get this data for device `dev1` in application `app1`:
+
+```bash
+$ curl --location --header "Authorization: Bearer NNSXS.XXXXXXXXX" --header 'Accept: application/json' \
+https://thethings.example.com/api/v3/applications/app1/devices/dev1?field_mask=name,description,locations
+```
+
+The same request in tenant `tenant1` on a multi-tenant deployment:
+
+```bash
+$ curl --location --header "Authorization: Bearer NNSXS.XXXXXXXXX" --header 'Accept: application/json' \
+https://tenant1.thethings.example.com/api/v3/applications/app1/devices/dev1?field_mask=name,description,locations
+```
+
+Fields may also be specified as a JSON object in a POST request.
+
+To get a stream of events for device `dev1` in application `app1` :
+
+```bash
+$ curl --location --header 'Authorization: Bearer NNSXS.XXXXXXXXX' --header 'Accept: application/json' \
+     --header 'Content-Type: application/json' --request POST 'https://thethings.example.com/api/v3/events' \
+     --data-raw '{
+    "identifiers":[{
+        "device_ids":{
+            "device_id":"dev1",
+            "application_ids":{"application_id":"app1"}
+        }
+    }]
+}'
+```
+
+If you want to create a device, perform multi step actions, or write shell scripts, it's best to use the [CLI]({{< ref "getting-started/cli" >}}).
+
+If you really, really want to do something like register a device manually, you need to make calls to the Identity Server, Join Server, Network Server and Application Server. 
+
+To register a device `newdev1` in application `app1`, first, register the `DevEUI`, `JoinEUI` and cluster addresses in the Identity Server:
+
+```bash
+$ curl --location \
+     --header 'Authorization: Bearer NNSXS.XXXXXXXXX' --header 'Content-Type: application/json' \
+     --request POST 'https://thethings.example.com/api/v3/applications/app1/devices' \
+     --data-raw '{
+      "end_device": {
+        "ids": {
+          "device_id": "newdev1",
+          "dev_eui": "0000000000000000",
+          "join_eui": "0000000000000000"
+        },
+        "join_server_address": "thethings.example.com",
+        "network_server_address": "thethings.example.com",
+        "application_server_address": "thethings.example.com"
+      },
+      "field_mask": {
+        "paths": [
+          "join_server_address",
+          "network_server_address",
+          "application_server_address",
+          "ids.dev_eui",
+          "ids.join_eui"
+        ]
+      }
+    }'
+```
+
+Register MAC settings in the Network Server:
```

Comment:
This doesn't really describe what's happening below, where we're not actually setting the `mac_setting`.